### ISSUE TITLE
ngModel that can't be parsed as a date causes infinite loop

### ISSUE THREAD

**Issue by isaacplmann** [vlio20/angular-datepicker#62]
Working on a PR.

**Comment by isaacplmann** [comment]
Trying to figure out how to reproduce in your demo first...

**Comment by vlio20** [comment]
Thanks. @isaacplmann,

**Comment by vlio20** [comment]
What undifined value you are talking about? For which property.

**Comment by isaacplmann** [comment]
I traced it down to day-calendar.service.ts line 42
```
    while (firstDayOfBoard.day() !== firstDayOfWeekIndex) {
      firstDayOfBoard.subtract(1, 'day');
    }
```

`firstDayOfBoard` is an invalid `Moment` object so subtracting a day returns another invalid `Moment` object.  In my app where the bug is happening, the `date` value for the top level `<dp-day-picker>` is `undefined`, but I'm not able to reproduce the same problem using your demo.

I'll keep looking and let you know what I find.

**Comment by vlio20** [comment]
@isaacplmann, as I mentioned I am working on big refactoring, which hopefully will simplify the extension of this module, you can follow the progress in the ref branch. Can you also check it there please?

**Comment by isaacplmann** [comment]
ok, I'll check in the ref branch

**Comment by isaacplmann** [comment]
ok, figured it out

If the top level `ngModel` value can't be parsed as a date, you'll hit the infinite loop.  So:
```
<dp-day-picker [ngModel]="'invalid'"></dp-day-picker>
```
The fix is to change line 36 of `day-calendar.service.ts` from:
```
const dayInMonth = config.month;
```
To this:
```
const dayInMonth = config.month.isValid() ? config.month : moment();
```

Should I submit a PR or wait until your refactoring is done and submit then?

**Comment by vlio20** [comment]
Better to wait. Thanks again.

**Comment by vlio20** [comment]
@isaacplmann, refactoring is done! Can you tell if the issue is still happening?

**Comment by isaacplmann** [comment]
Looks like it's fixed.  Thanks!  When do you think you'll publish to npm?

**Comment by vlio20** [comment]
Today or tomorrow.

**Comment by vlio20** [comment]
@isaacplmann, the release is available under 1.1.0 - please note that there are many breaking changes. Please let me know if you find any issues.
